pkg/kine/drivers/generic/generic.go

```diff
@@ -394,15 +396,32 @@ func (d *Generic) queryRowPrepared(ctx context.Context, txName, sql string, prep
 }
 
 func (d *Generic) InitializeWriteBatching(ctx context.Context) {
-	if d.BatchingEnabled && d.BatchingInterval > 0 && d.BatchingMaxQueries > 0 {
-		d.flushCh = make(chan struct{})
-		d.addToBatchQueueCh = make(chan BatchedInsert)
-		d.removeFromBatchQueueCh = make(chan int)
-		d.batchingQueue = make([]BatchedInsert, 0, d.BatchingMaxQueries)
 
-		go d.watchBatchQueue(ctx)
-		go d.batchQueueWatcher(ctx)
+	if !d.BatchingEnabled {
+		return
 	}
+
+	if d.BatchingInterval < 10*time.Millisecond {
```

Comment:
It would make sense to give an upper limit here as well to:
Otherwise inexperienced users might set high intervals which might block the calls for too long.